src/Aspire.Dashboard/Components/Controls/LogViewer.razor

```diff
@@ -6,7 +6,27 @@
 @inject IJSRuntime JS
 @implements IAsyncDisposable
 
-<div class="log-overflow">
+<div class="log-overflow continuous-scroll-overflow">
     <div class="log-container" id="logContainer">
+        <Virtualize Items="_logEntries" ItemSize="20" OverscanCount="20">
+            <div class="line-row-container">
+                <div class="line-row">
+                    <span class="line-area" role="log">
+                        <span class="line-number">@(_logEntries.IndexOf(context) + 1)</span>
```

Comment:
This is going to end up scanning the whole log entry any time you're scrolled to the bottom, isn't it? This is probably part of what you were referencing in your description. Wonder if we could track the index on the item itself. Would require changing ones that are already in the list as we go past them to insert another, but that's not _too_ hard.

Reply:
Yes, this is a problem for lots of reasons. Still TODO.